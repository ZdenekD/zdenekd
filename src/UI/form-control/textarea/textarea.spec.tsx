import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../../__test__/utils/helpers';
import Textarea from '.';

describe('Textarea', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Textarea name="textarea" label="Textarea" />);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-textarea');

        expect(component.exists()).toBe(true);
    });

    it('set length on change', () => {
        const mockSetState = jest.fn();
        const value = 'value';
        const mockEvent = {target: {value}};

        React.useState = jest.fn(() => [0, mockSetState]);

        const container = shallow(<Textarea name="textarea" label="textarea" />);
        const component = findComponent(container, 'component-textarea');

        component.simulate('change', mockEvent);

        expect(mockSetState).toHaveBeenCalledWith(value.length);
    });
});
