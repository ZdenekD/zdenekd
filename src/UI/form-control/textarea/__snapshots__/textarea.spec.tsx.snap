// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea match snapshot 1`] = `
<div class="control   ">
  <textarea id="_34rFI1"
            name="textarea"
            class="textarea   "
            data-test="component-textarea"
  >
  </textarea>
  <label for="_34rFI1"
         class="label"
  >
    <span class="labelContent">
      Textarea
    </span>
  </label>
</div>
`;
